### PR TITLE
doc: Improve gui/src/qt README.md

### PR DESCRIPTION
**Pull request by jarolrod**
**Master/Before:** [Render of Master](https://github.com/bitcoin-core/gui/blob/master/src/qt/README.md)

**PR/After:** [Render of PR](https://github.com/bitcoin-core/gui/blob/5d1f260713f9f1d29c0db68f2917dfe7368d4ba0/src/qt/README.md)

**Changes:**
The README.md found in `gui/src/qt` seems to not have gotten any love in a while. This PR fixes some grammatical errors, makes it easier to follow, and modernizes the logic of using Qt Creator. 
1. Makes several sections more informative
2. Directories under `Files and Directories` now end with a forward slash denoting that they are a directory
3. Modernize the Qt Creator Logic for the current setup flow
4. Add UNIX Qt Creator Setup Instructions (Ubuntu & Debian)

### CODE REVIEW

**Review by hebasto — COMMENTED** [review]
For editing `*.ui` files the `Qt Designer` is enough for me, but this opinion is personal, ofc.

File: src/qt/README.md [hunk 1]
```diff
@@ -1,62 +1,61 @@
-This directory contains the BitcoinQT graphical user interface (GUI). It uses the cross-platform framework [Qt](https://www1.qt.io/developers/).
+This directory contains the source code for the reference implementation of the Bitcoin Core graphical user interface (GUI). It uses the [Qt](https://www1.qt.io/developers/) cross-platform framework.
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md:1]: I was drawing on conversations had about the role of the Bitcoin Core GUI. Here is an example https://github.com/bitcoin-core/gui/issues/45#issuecomment-671333460. I think it's OK to call this the reference implementation. If it's not OK with the community, I'll change this to: `This directory contains the source code for the Bitcoin Core graphical user interface (GUI).`

File: src/qt/README.md [hunk 2]
```diff
@@ -1,62 +1,61 @@
...
 
 ```sh
 ./src/qt/bitcoin-qt
 ```
 
-## Files and directories
+## Files and Directories
 
-### forms
+#### /forms
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md:17]: You're right, `forms/` is more appropriate

File: src/qt/README.md [hunk 3]
```diff
@@ -1,62 +1,61 @@
...
 
-Contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. They are created with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
+- A directory which contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. These files specify the characteristics of form elements in XML. Qt UI files are normally altered with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
 
-### locale
+#### /locale
 
-Contains translations. They are periodically updated. The process is described [here](/doc/translation_process.md).
+- Contains translations of the strings used throughout the GUI. They are periodically updated and an effort is made to support as many languages as possible. The process of contributing translations is described [here](/doc/translation_process.md).
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md:23]: gui/src/qt/locale/ seems to only contain translations for the GUI

File: src/qt/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (RandyMcMillan) [reply src/qt/README.md:1]: I like this better..

`
This directory contains the source code for the Bitcoin Core graphical user interface (GUI).
`

It is more correct IMO.

File: src/qt/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hebasto) [reply src/qt/README.md:23]: https://github.com/bitcoin-core/gui/blob/7ae86b3c6845873ca96650fc69beb4ae5285c801/src/qt/locale/bitcoin_en.ts#L4547

https://github.com/bitcoin-core/gui/blob/7ae86b3c6845873ca96650fc69beb4ae5285c801/src/init.cpp#L1127

File: src/qt/README.md [hunk 4]
```diff
@@ -1,62 +1,61 @@
...
+#### \*dialog.(h/cpp)
 
-### \*dialog.(h/cpp)
+- Various dialogs, e.g. to open a URL. Inherit from [QDialog](https://doc.qt.io/qt-5/qdialog.html).
 
-Various dialogs, e.g. to open a URL. Inherit from [QDialog](https://doc.qt.io/qt-5/qdialog.html).
+#### paymentserver.(h/cpp)
 
-### paymentserver.(h/cpp)
+- Used to process BIP21 payment URI requests. Also handles URI based application switching (e.g. when following a bitcoin:... link from a browser).
```
Comment (jonasschnelli) [inline src/qt/README.md:52]: Maybe add `(deprecated)`?

File: src/qt/README.md [hunk 5]
```diff
@@ -75,21 +74,23 @@ See [CONTRIBUTING.md](/CONTRIBUTING.md) for general guidelines. Specifically for
...
+1. Make sure you've installed all dependencies specified in your systems build instructions
+2. Follow the compile instructions for your system, run `./configure` with the `--enable-debug` flag
+3. Start Qt Creator. At the start page, do: `New` -> `Import Project` -> `Import Existing Project`
+5. Enter `"bitcoin-qt"` as the Project Name, enter `"{repo}/src/qt"` as Location (replace `{repo}` with the actual repo path)
+6. Check over the file selection, you may need to select the `"forms"` directory (necessary if you intend to edit *.ui files)
+7. Confirm the `Summary` page
+8. In the `Projects` tab, select `Manage Kits...`
+ - Under `Kits`: select the default "Desktop" kit
+ - Under `Compilers`: select an appropriate compiler. (on macOS select `"Clang (x86 64bit in /usr/bin)"`)
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
```
Comment (jonasschnelli) [inline src/qt/README.md:93]: I guess this is Mac specific (clang in /usr/bin/ as well as `LLDB`). I would probably keep this as "Mac" instructions (Linux/Win) people might need to adapt but they are not blinded by not telling them its made for Mac.

**Comment by jonasschnelli** [comment]
ACK modulo 2xcomments

**Comment by RandyMcMillan** [comment]
@jarolrod - You should be able to trigger a rebuild yourself.
https://cirrus-ci.com/task/6120245750398976

ACK :)

**Comment by RandyMcMillan** [comment]
The parent commits never passed CI

The current parent commits

https://github.com/bitcoin-core/gui/commit/ca759bc743c9b3b240289c467349cefc5f4444af
https://github.com/bitcoin-core/gui/commit/62c93c34c1d9b83cd168b6b1e30d10e01db53e38
https://github.com/bitcoin-core/gui/commit/b40d4d494e1e87fecdba05463430749971c1b023

I suggest rebasing it to 

https://github.com/bitcoin-core/gui/commit/a0489f3472f3799dc1ece32a59556fd239c4c14b

As you can see - it passed CIRRUS

https://cirrus-ci.com/build/4908623950249984

and travis-ci

https://travis-ci.org/github/bitcoin-core/gui/builds/747420647

File: src/qt/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md]: Addressed this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Used @RandyMcMillan suggestion, thanks!

File: src/qt/README.md [hunk 6]
```diff
@@ -1,62 +1,61 @@
-This directory contains the BitcoinQT graphical user interface (GUI). It uses the cross-platform framework [Qt](https://www1.qt.io/developers/).
+This directory contains the source code for the reference implementation of the Bitcoin Core graphical user interface (GUI). It uses the [Qt](https://www1.qt.io/developers/) cross-platform framework.
 
 The current precise version for Qt 5 is specified in [qt.mk](/depends/packages/qt.mk).
 
 ## Compile and run
 
-See build instructions ([macOS](/doc/build-osx.md), [Windows](/doc/build-windows.md), [Unix](/doc/build-unix.md), etc).
+See build instructions ([macOS](/doc/build-osx.md), [Windows](/doc/build-windows.md), [Unix](/doc/build-unix.md), etc). Compiling generates the `bitcoin-qt` executable.
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md]: Addresses this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Removed this string in favor of just referring users to the build instruction.

File: src/qt/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qt/README.md]: addressed this in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304. Removed the `gui` specific part.

File: src/qt/README.md (hunk 4 above)
Comment (jarolrod) [reply src/qt/README.md]: Done in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

File: src/qt/README.md (hunk 5 above)
Comment (jarolrod) [reply src/qt/README.md:93]: Done in f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

**Comment by jarolrod** [comment]
updated to f6584d9c84f4d840dd5cb49f7dcc9984e9a16304

Addressed @hebasto [comments](https://github.com/bitcoin-core/gui/pull/139#pullrequestreview-540386515) and @jonasschnelli [suggestions](https://github.com/bitcoin-core/gui/pull/139#issuecomment-736355489)

File: src/qt/README.md [hunk 7]
```diff
@@ -1,62 +1,61 @@
...
 ```
 
-## Files and directories
+## Files and Directories
 
-### forms
+#### forms/
 
-Contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. They are created with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
+- A directory which contains [Designer UI](https://doc.qt.io/qt-5.9/designer-using-a-ui-file.html) files. These files specify the characteristics of form elements in XML. Qt UI files are normally altered with [Qt Creator](#using-qt-creator-as-ide), but can be edited using any text editor.
```
Comment (jonatack) [inline src/qt/README.md:19]: - s/which/that/
- s/normally altered/can be edited/
- s/, but can be edited/or/

**Review by jonatack — COMMENTED** [review]
Concept ACK

**Comment by jarolrod** [comment]
updated f6584d9 -> a9e641d

Implemented @jonatack suggestions. Thanks for the in-depth review!

Additionally, I have included some Qt Creator setup instructions for UNIX (Ubuntu & Debian). Windows instructions still need to be done. I will take care of that in a follow-up PR.

New Render: [src/qt/README.md](https://github.com/bitcoin-core/gui/blob/a9e641d40d83f1a89bc57ab14f39ed370e0c7e2a/src/qt/README.md)

**Comment by jonatack** [comment]
Thanks, will re-review.

File: src/qt/README.md [hunk 8]
```diff
@@ -69,27 +70,54 @@ Represents the view to a single wallet.
...
+5. Check over the file selection, you may need to select the `forms` directory (necessary if you intend to edit *.ui files)
+6. Confirm the `Summary` page
+7. In the `Projects` tab, select `Manage Kits...`
 
-You can use Qt Creator as an IDE. This is especially useful if you want to change
-the UI layout.
+ **macOS**
+ - Under `Kits`: select the default "Desktop" kit
+ - Under `Compilers`: select `"Clang (x86 64bit in /usr/bin)"`
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
```
Comment (jonatack) [inline src/qt/README.md:112]: remove whitespace at EOL
```diff
  - Under `Compilers`: select `"Clang (x86 64bit in /usr/bin)"`
- - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)  
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
 
- **Ubuntu & Debian**  
+ **Ubuntu & Debian**
  **Note:** Some of these options may already be set
```

File: src/qt/README.md [hunk 9]
```diff
@@ -1,62 +1,63 @@
...
+- (Deprecated) Used to process BIP21 payment URI requests. Also handles URI-based application switching (e.g. when following a bitcoin:... link from a browser).
 
-### walletview.(h/cpp)
+#### walletview.(h/cpp)
 
-Represents the view to a single wallet.
+- Represents the view to a single wallet.
 
-### Other .h/cpp files
+#### Other .h/cpp files
```
Comment (jonatack) [inline src/qt/README.md:60]: lines 64 and 66: s/etc/etc./

**Review by jonatack — COMMENTED** [review]
Almost-ACK, just a couple more items below.

**Comment by jarolrod** [comment]
updated a9e641d -> 63d0f29c9945dcf512bf494a4f24190121c48648, addressed @jonatack [suggestions](https://github.com/bitcoin-core/gui/pull/139#pullrequestreview-573287949)

File: src/qt/README.md [hunk 10]
```diff
@@ -1,95 +1,123 @@
...
-You can use Qt Creator as an IDE. This is especially useful if you want to change
-the UI layout.
+ **macOS**
+ - Under `Kits`: select the default "Desktop" kit
+ - Under `Compilers`: select `"Clang (x86 64bit in /usr/bin)"`
+ - Under `Debuggers`: select `"LLDB"` as debugger (you might need to set the path to your LLDB installation)
 
-Download and install the community edition of [Qt Creator](https://www.qt.io/download/).
-Uncheck everything except Qt Creator during the installation process.
+ **Ubuntu & Debian**
```
Comment (jonatack) [inline src/qt/README.md:114]:
```suggestion
 **Ubuntu & Debian**
```

Comment (jarolrod) [reply src/qt/README.md]: addressed in 5d1f260

**Comment by jarolrod** [comment]
updated 63d0f29c9945dcf512bf494a4f24190121c48648 -> 5d1f260, addressed leading whitespace issue

**Review by jonatack — APPROVED** [review]
ACK 5d1f260713f9f1d29c0db68f2917dfe7368d4ba0

**Comment by RandyMcMillan** [comment]
ACK 👍🏼

**Pull request merged**